<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>707</width>
    <height>486</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLineEdit" name="inputLine">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>651</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>150</y>
      <width>651</width>
      <height>321</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="1">
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_calculus">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>∫(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_derivative">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>ｄ／ｄｘ(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_assign">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>&lt;-</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pushButton_undo">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Undo</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="pushButton_redo">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Redo</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButton_root">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>root(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_sigma">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>∑(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_lambda">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>λ(x){}</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_x">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>x</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="2">
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_C">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_C</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_A">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_A</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_B">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_B</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_D">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_D</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_E">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_E</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_F">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>_F</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="1">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_sub">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>-</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_mult">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>*</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_ac">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>AC</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_rightparen">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_div">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>/</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_add">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>+</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_leftparen">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>(</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pushButton_ans">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Ans</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="pushButton_comma">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>,</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButton_mod">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="pushButton_eval">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>=</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_del">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Del</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_7">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>7</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_9">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>9</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_1">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>1</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="pushButton_3">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>3</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_2">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>2</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="pushButton_dot">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>.</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_8">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>8</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_6">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>6</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_4">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>4</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButton_0">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="pushButton_10e">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>E</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_5">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>5</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="0">
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="4" column="1">
        <widget class="QPushButton" name="pushButton_lg">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>lg(</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="pushButton_ln">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>ln(</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="pushButton_log">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>log(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_sin">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>sin(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_tan">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>tan(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="pushButton_cos">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>cos(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_asin">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>asin(</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QPushButton" name="pushButton_acos">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>acos(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_atan">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>atan(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="pushButton_sinh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>sinh(</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="pushButton_asinh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>asinh(</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="pushButton_tanh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>tanh(</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QPushButton" name="pushButton_atanh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>atanh(</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_cosh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>cosh(</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_acosh">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>acosh(</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QPushButton" name="pushButton_pow">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>pow(</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="2">
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButton_e">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>ｅ</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButton_advance">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Advance</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_pi">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>π</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QTableWidget" name="outputList">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>651</width>
      <height>101</height>
     </rect>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::CustomContextMenu</enum>
    </property>
    <property name="columnCount">
     <number>2</number>
    </property>
    <column>
     <property name="text">
      <string>表达式</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>值</string>
     </property>
    </column>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_ac</sender>
   <signal>clicked()</signal>
   <receiver>inputLine</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>413</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_undo</sender>
   <signal>clicked()</signal>
   <receiver>inputLine</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>422</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>391</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_redo</sender>
   <signal>clicked()</signal>
   <receiver>inputLine</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>503</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
